### PR TITLE
Unique list refactor

### PR DESCRIPTION
**Pull request by Badatprogrammiing**
UniqueList is now a generic, Elements of UniqueList have to implement Unique Interface. Add CLI tabs switching and  internship models...

### CODE REVIEW

File: src/main/java/seedu/address/model/fields/Outcome.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Represents the possible outcome for an application/task
+ */
+public enum Outcome {
+    FOLLOW_UP, GHOSTED, REJECTED, OFFERED, ACCEPTED;
+
+    @Override
+    public String toString() {
+        switch (this) {
+        case FOLLOW_UP:
+            return "follow up";
```
Comment (seewhyjay) [inline src/main/java/seedu/address/model/fields/Outcome.java:13]: Capitalise all and change to Follow-up

File: src/main/java/seedu/address/model/fields/Outcome.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Represents the possible outcome for an application/task
+ */
+public enum Outcome {
+    FOLLOW_UP, GHOSTED, REJECTED, OFFERED, ACCEPTED;
+
+    @Override
+    public String toString() {
+        switch (this) {
```
Comment (seewhyjay) [inline src/main/java/seedu/address/model/fields/Outcome.java:11]: maybe switch (this.toLowerCase())?

File: src/main/java/seedu/address/model/fields/ApplicationOutcome.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * An outcome for a job application
+ */
+public class ApplicationOutcome {
+
+    private static final List<String> possibleOutcomes =
+            new LinkedList<>(Arrays.asList("follow up", "ghosted", "offered", "rejected", "accepted"));
```
Comment (seewhyjay) [inline src/main/java/seedu/address/model/fields/ApplicationOutcome.java:15]: is there a reason for this to be a LinkedList?

File: src/main/java/seedu/address/model/fields/Outcome.java (hunk 1 above)
Comment (Badatprogrammiing) [reply src/main/java/seedu/address/model/fields/Outcome.java:13]: ok

File: src/main/java/seedu/address/model/fields/Outcome.java (hunk 2 above)
Comment (Badatprogrammiing) [reply src/main/java/seedu/address/model/fields/Outcome.java:11]: FOLLOW_UP got underscore, cannot directly covert to Follow-up, abit laze, so i just handle all cases

File: src/main/java/seedu/address/model/fields/ApplicationOutcome.java (hunk 3 above)
Comment (Badatprogrammiing) [reply src/main/java/seedu/address/model/fields/ApplicationOutcome.java:15]: Hmm, what do u suggest?

Comment (seewhyjay) [reply src/main/java/seedu/address/model/fields/ApplicationOutcome.java:15]: nah, just wondering if there was a function of linkedlists you need to use

Comment (seewhyjay) [inline src/main/java/seedu/address/model/fields/ApplicationOutcome.java:15]: does this need to be caps and `Follow-up` for consistency?

File: src/main/java/seedu/address/model/fields/Outcome.java (hunk 2 above)
Comment (seewhyjay) [reply src/main/java/seedu/address/model/fields/Outcome.java:11]: mb I was thinking of the parser, but for the returns can capitalise first letter and `Follow-up` here as well?

**Review by seewhyjay — APPROVED** [review]
thanks for carry

**Pull request merged**